CHANGELOG.md

```diff
@@ -3,6 +3,7 @@
 ### Breaking changes
 
 * `RealmResults.distinct()` returns a new `RealmResults` object instead of filtering on the original object (#2947).
+* `RealmResults` is auto-updated continuously. Any transaction on the caller thread which may have an impact on the order or elements of the `RealmResults` will change the `RealmResults` immediately instead of change it in the next event loop. Iterator behavior of `RealmResults` stays the same, transactions inside the iterating still works as expected.
```

Comment:
Something like that yes, but we still need to be very clear when it now breaks. Perhaps "The standard `RealmResults.iterator()` will continue to work as normal, which means that you can still delete or modify elements without impacting the iterator. The same is not true for simple for loops. In some cases a simple for loop will not work (link-to-example), and you must either iterate backwards (link-to-example) or use the new `createSnapshot()` method."

What do you think?